app-modules/engagement/database/migrations/2025_04_17_142824_data_migration_refactor_engagement_subject_to_jsonb.php

```diff
@@ -0,0 +1,224 @@
+                foreach ($rows as $row) {
+                    /** @var array<string, mixed> $decoded */
+                    $decoded = json_decode($row->subject, true);
+
+                    /** @var array<int, array{type: string, content?: array<int, array{type: string, text: string}>}> $blocks */
+                    $blocks = $decoded['content'] ?? [];
+
+                    /** @var string $text */
+                    $text = collect($blocks)
+                        ->flatMap(fn ($block) => collect($block['content'] ?? []))
+                        ->filter(fn ($node) => $node['type'] === 'text')
+                        ->pluck('text')
+                        ->implode('');
+
+                    DB::table('engagements')
+                        ->where('id', $row->id)
+                        ->update(['subject' => $text]);
+                }
+            });
+
+            DB::table('engagement_batches')->select('id', 'subject')->chunkById(500, function ($rows) {
+                foreach ($rows as $row) {
+                    /** @var array<string, mixed> $decoded */
+                    $decoded = json_decode($row->subject, true);
+
+                    /** @var array<int, array{type: string, content?: array<int, array{type: string, text: string}>}> $blocks */
+                    $blocks = $decoded['content'] ?? [];
+
+                    /** @var string $text */
+                    $text = collect($blocks)
+                        ->flatMap(fn ($block) => collect($block['content'] ?? []))
+                        ->filter(fn ($node) => $node['type'] === 'text')
+                        ->pluck('text')
+                        ->implode('');
+
+                    DB::table('engagement_batches')
+                        ->where('id', $row->id)
+                        ->update(['subject' => $text]);
+                }
+            });
+
+            DB::table('form_email_auto_replies')->select('id', 'subject')->chunkById(500, function ($rows) {
+                foreach ($rows as $row) {
+                    /** @var array<string, mixed> $decoded */
+                    $decoded = json_decode($row->subject, true);
+
+                    /** @var array<int, array{type: string, content?: array<int, array{type: string, text: string}>}> $blocks */
+                    $blocks = $decoded['content'] ?? [];
+
+                    /** @var string $text */
+                    $text = collect($blocks)
+                        ->flatMap(fn ($block) => collect($block['content'] ?? []))
+                        ->filter(fn ($node) => $node['type'] === 'text')
+                        ->pluck('text')
+                        ->implode('');
+
+                    DB::table('form_email_auto_replies')
+                        ->where('id', $row->id)
+                        ->update(['subject' => $text]);
+                }
+            });
+
+            DB::table('campaign_actions')
```

Comment:
Same as above, we we can ensure this is only happening on ones that are engagements.